src/pages/library/datasetBuilder/DatasetBuilder.test.ts

```diff
@@ -0,0 +1,154 @@
+import { fireEvent, render } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import _ from 'lodash/fp';
+import { h } from 'react-hyperscript-helpers';
+import * as Nav from 'src/libs/nav';
+import { dataCatalogStore, datasetBuilderCohorts, datasetBuilderConceptSets } from 'src/libs/state';
+import { PREPACKAGED_CONCEPT_SETS } from 'src/pages/library/datasetBuilder/constants';
+import { Cohort, ConceptSet } from 'src/pages/library/datasetBuilder/dataset-builder-types';
+import {
+  CohortSelector,
+  ConceptSetSelector,
+  CreateCohortModal,
+  DatasetBuilderContents,
+  ValuesSelector,
+} from 'src/pages/library/datasetBuilder/DatasetBuilder';
+
+jest.mock('src/libs/nav', () => ({
+  ...jest.requireActual('src/libs/nav'),
+  getLink: jest.fn(),
+  useRoute: jest.fn(),
+}));
+
+type ModalMockExports = typeof import('src/components/Modal.mock');
+jest.mock('src/components/Modal', () => {
+  const mockModal = jest.requireActual<ModalMockExports>('src/components/Modal.mock');
+  return mockModal.mockModalModule();
+});
```

Comment:
Is there a reason we can't directly import `mockModalModule` like in [this test](https://github.com/DataBiosphere/terra-ui/blob/f3d9033bff35c429640d490bb935fad776ed4caf/src/components/RequesterPaysModal.test.js#L8)?

Reply:
For some reason that seems not to work. I was noticing that test is JS rather than TS, which might be the difference? I was copying the pattern in [this test](https://github.com/DataBiosphere/terra-ui/blob/dev/src/pages/workspaces/workspace/analysis/modals/AnalysisDuplicator.test.ts#L26)

Reply:
the difference is in fact TS versus JS.  TS is forcing better (and helpful) type safety guard-rails.   Because of jest mock hoisting restrictions/mechanics, the pattern above is the recommended way to a) make Typescript happy, and b) make the mock safer to extend/modify in the future, and c) provide go-to-definition tooling etc. to the shared mock being pulled in.